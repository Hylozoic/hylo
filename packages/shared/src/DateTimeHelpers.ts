import { DateTime, DateTimeUnit } from 'luxon'
import prettyDate from 'pretty-date'

export const getLocaleAsString = (locale : string ) : string => {
  switch (locale) {
    case 'en':
      return 'en-US'
    case 'es':
      return 'es'
    default:
      return 'en-US'
  }
}

export const dateTimeNow = (locale?: string) : DateTime => {
  return DateTime.now().setLocale(getLocaleAsString(locale || ''))
}

export interface ToDateTimeOptions {
  timezone?: string;
  locale?: string;
}

export const toDateTime = (
  dt: string | Date | DateTime | Object,
  options?: ToDateTimeOptions
): DateTime => {
  const { timezone, locale } = options || {};
  const zoneOption = { zone: timezone || DateTime.now().zoneName || 'UTC' }
  const _dt = dt instanceof DateTime
    ? dt
    : dt instanceof Date
      ? DateTime.fromJSDate(dt, zoneOption).setLocale(getLocaleAsString(locale || ''))
      : typeof dt === 'string'
        ? DateTime.fromISO(dt, zoneOption).setLocale(getLocaleAsString(locale || ''))
        : DateTime.fromObject(dt, zoneOption).setLocale(getLocaleAsString(locale || ''))
  if (_dt.invalidReason) {
    throw new Error(`Invalid date: ${_dt.invalidReason}`)
  }
  return _dt
}

const isSame = (
  dt1: string | Date | DateTime | Object,
  dt2: string | Date | DateTime | Object,
  unit: DateTimeUnit,
  locale?: string
): boolean => {
  const _dt1 = toDateTime(dt1, { locale })
  const _dt2 = toDateTime(dt2, { locale })
  return _dt1.hasSame(_dt2, unit)
}

export const rangeIncludesDate = (
  start: string | Date | DateTime | Object,
  date: string | Date | DateTime | Object,
  end: string | Date | DateTime | Object,
): boolean => {
  const _start = toDateTime(start)
  const _date = toDateTime(date)
  const _end = toDateTime(end)
  return _date.hasSame(_start, 'day')
    || _date.hasSame(_end, 'day')
    || (_start < _date && _date < _end)
}

export const inWeek = (
  start: string | Date | DateTime | Object,
  date: string | Date | DateTime | Object,
  end: string | Date | DateTime | Object
): boolean => {
  const _start = toDateTime(start)
  const _date = toDateTime(date)
  const _end = toDateTime(end)
  const weekStart = _date.startOf('week', { useLocaleWeeks: true })
  const weekEnd = _date.endOf('week', { useLocaleWeeks: true }).plus({ days: 1 })
  return _start < weekEnd && weekStart <= _end
}

export const isSameDay = (
  // TODO datetime: need to check if these dates already have locale set
  date1 : string | Date | DateTime | Object,
  date2 : string | Date | DateTime | Object
) : boolean => {
  return isSame(date1, date2, 'day')
}

export const isSameWeek = (
  date1 : string | Date | DateTime | Object,
  date2 : string | Date | DateTime | Object
) : boolean => {
  return isSame(date1, date2, 'week')
}

export const isSameMonth = (
  date1 : string | Date | DateTime | Object,
  date2 : string | Date | DateTime | Object
) : boolean => {
  return isSame(date1, date2, 'month')
}

export function humanDate (
  date: string | Date | DateTime | Object,
  short?: boolean,
  locale?: string
): string {
  const _date = toDateTime(date, { locale })
  let ret = _date.invalidReason ? '' : prettyDate.format(_date.toJSDate())

  // Always return 'now' for very recent timestamps
  if (ret === 'just now') {
    return 'now'
  }

  if (short) {
    ret = ret.replace(' ago', '')
  } else {
    if (ret.match(/(\d+) seconds? ago/)) {
      return 'now'
    }
  }

  return ret.replace(/ seconds?/, 's')
    .replace(/ minutes?/, 'm')
    .replace(/ hours?/, 'h')
    .replace(/ days?/, 'd')
    .replace(/ weeks?/, 'w')
    .replace(/ years?/, 'y')
    .replace(/ month(s?)/, ' mo$1')
}

export const formatDatePair = (
  start: string | Date | DateTime | Object,
  end: string | Date | DateTime | Object | boolean,
  returnAsObj?: boolean,
  timezone?: string,
  locale?: string
): string | { from: string, to: string } => {
  const _start = toDateTime(start, { timezone, locale })
  const _end = end ? toDateTime(end, { timezone, locale }) : null

  const now = dateTimeNow(locale)

  const isPastYear = _start.get('year') < now.get('year')
  const isSameDay = _end && _start.get('day') === _end.get('day') &&
                    _start.get('month') === _end.get('month') &&
                    _start.get('year') === _end.get('year')

  let to = ''
  let from = ''

  // Format the start date - only include year if it's in the past
  if (isPastYear) {
    from = _start.toFormat("ccc MMM d, yyyy '•' t")
  } else {
    from = _start.toFormat("ccc MMM d '•' t")
  }

  // Format the end date/time if provided
  if (_end) {
    if (isSameDay) {
      // If same day, only show the end time
      to = _end.toFormat('t')
    } else if (_end.get('year') < now.get('year')) {
      // If end date is in a past year, include the year
      to = _end.toFormat("MMM d, yyyy '•' t")
    } else {
      // Otherwise just month, day and time
      to = _end.toFormat("MMM d '•' t")
    }

    to = returnAsObj ? to : ' - ' + to
  }

  return returnAsObj ? { from, to } : from + to
}

export function isDateInTheFuture (
  // TODO datetime: need to check if this date already has locale set
  date: string | Date | DateTime | Object,
  locale?: string
): boolean {
  return toDateTime(date, { locale }) > dateTimeNow(locale)
}

/**
 * Returns the month name from a date string or Date object
 * @param {string|Date} date - Date string or Date object
 * @param {boolean} short - Whether to return short month name (e.g. 'Jan' vs 'January')
 * @param {string} timezone - Optional timezone (defaults to local timezone)
 * @param {string} locale - Optional locale for formatting
 * @returns {string} Month name
 */
export function getMonthFromDate (
  date: string | Date | DateTime | Object,
  short?: boolean,
  timezone?: string,
  locale?: string
): string {
  return toDateTime(date, { timezone, locale }).toFormat(short ? 'MMM' : 'MMMM')
}

/**
 * Returns the day number from a date string or Date object
 * @param {string|Date} date - Date string or Date object
 * @param {string} timezone - Optional timezone (defaults to local timezone)
 * @param {string} locale - Optional locale for formatting
 * @returns {number} Day of month (1-31)
 */
export function getDayFromDate (
  date: string | Date | DateTime | Object,
  timezone?: string,
): number {
  return toDateTime(date, { timezone }).day
}

/**
 * Returns the hour from a date string or Date object
 * @param {string|Date} date - Date string or Date object
 * @param {boolean} use24Hour - Whether to use 24-hour format (default: false)
 * @param {string} timezone - Optional timezone (defaults to local timezone)
 * @param {string} locale - Optional locale for formatting
 * @returns {string} Formatted hour (with AM/PM if use24Hour is false)
 */
export function getHourFromDate (
  date: string | Date | DateTime | Object,
  use24Hour?: boolean,
  timezone?: string
): string {
  return toDateTime(date, { timezone }).toFormat(use24Hour ? 'HH' : 'h a')
}
