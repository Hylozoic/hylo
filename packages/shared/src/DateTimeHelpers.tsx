import { DateTime, DateTimeUnit } from 'luxon'
import { localeLocalStorageSync } from '../../../apps/web/src/util/locale'
import prettyDate from 'pretty-date'

export const getLocaleAsString = () : string => {
  switch (localeLocalStorageSync()) {
    case 'en':
      return 'en-US'
    case 'es':
      return 'es'
    default:
      return 'en-US'
  }
}

export const dateTimeNow = () : DateTime => {
  return DateTime.now().setLocale(getLocaleAsString())
}

export const toDateTime = (
  dt : string | Date | DateTime | Object,
  timezone? : string
) : DateTime => {
  const zoneOption = { zone: timezone || DateTime.now().zoneName || 'UTC' }
  const _dt = dt instanceof DateTime
    ? dt
    : dt instanceof Date
      ? DateTime.fromJSDate(dt, zoneOption).setLocale(getLocaleAsString())
      : typeof dt === 'string'
        ? DateTime.fromISO(dt, zoneOption).setLocale(getLocaleAsString())
        : DateTime.fromObject(dt, zoneOption).setLocale(getLocaleAsString())
  if (_dt.invalidReason) {
    throw new Error(`Invalid date: ${_dt.invalidReason}`)
  }
  return _dt
}

const same = (
  dt1 : string | Date | DateTime | Object,
  dt2 : string | Date | DateTime | Object,
  unit : DateTimeUnit
) : boolean => {
  const _dt1 = toDateTime(dt1)
  const _dt2 = toDateTime(dt2)
  return _dt1.hasSame(_dt2, unit)
}

export const includes = (
  dt1 : string | Date | DateTime | Object,
  dt2 : string | Date | DateTime | Object,
  dt3 : string | Date | DateTime | Object
) : boolean => {
  const _dt1 = toDateTime(dt1)
  const _dt2 = toDateTime(dt2)
  const _dt3 = toDateTime(dt3)
  return _dt2.hasSame(_dt1, 'day')
    || _dt2.hasSame(_dt3, 'day')
    || (_dt1 <= _dt2 && _dt2 < _dt3)
}

export const inWeek = (
  dt1 : string | Date | DateTime | Object,
  dt2 : string | Date | DateTime | Object,
  dt3 : string | Date | DateTime | Object
) : boolean => {
  const _dt1 = toDateTime(dt1)
  const _dt2 = toDateTime(dt2)
  const _dt3 = toDateTime(dt3)
  const weekStart = _dt2.startOf('week', { useLocaleWeeks: true })
  const weekEnd = _dt2.endOf('week', { useLocaleWeeks: true }).plus({ days: 1 })
  return _dt1 < weekEnd && weekStart <= _dt3
}

export const sameDay = (
  dt1 : string | Date | DateTime | Object,
  dt2 : string | Date | DateTime | Object,
  dt3? : string | Date | DateTime | Object
) : boolean => {
  return dt3 ? includes(dt1, dt2, dt3) : same(dt1, dt2, 'day')
}

export const sameWeek = (
  dt1 : string | Date | DateTime | Object,
  dt2 : string | Date | DateTime | Object
) : boolean => {
  return same(dt1, dt2, 'week')
}

export const sameMonth = (
  dt1 : string | Date | DateTime | Object,
  dt2 : string | Date | DateTime | Object
) : boolean => {
  return same(dt1, dt2, 'month')
}

export function humanDate (
  dt : string | Date | DateTime | Object,
  short? : boolean
) : string {
  const _dt = toDateTime(dt)
  let ret = _dt.invalidReason ? '' : prettyDate.format(_dt.toJSDate())

  // Always return 'now' for very recent timestamps
  if (ret === 'just now') {
    return 'now'
  }

  if (short) {
    ret = ret.replace(' ago', '')
  } else {
    if (ret.match(/(\d+) seconds? ago/)) {
      return 'now'
    }
  }

  return ret.replace(/ seconds?/, 's')
    .replace(/ minutes?/, 'm')
    .replace(/ hours?/, 'h')
    .replace(/ days?/, 'd')
    .replace(/ weeks?/, 'w')
    .replace(/ years?/, 'y')
    .replace(/ month(s?)/, ' mo$1')
}

export const formatDatePair = (
  startTime : string | Date | DateTime | Object,
  endTime : string | Date | DateTime | Object | boolean,
  returnAsObj? : boolean,
  timezone? : string
) : string | { from : string, to : string } => {
  const start = toDateTime(startTime, timezone)
  const end = endTime ? toDateTime(endTime, timezone) : null

  const now = dateTimeNow()

  const isPastYear = start.get('year') < now.get('year')
  const isSameDay = end && start.get('day') === end.get('day') &&
                    start.get('month') === end.get('month') &&
                    start.get('year') === end.get('year')

  let to = ''
  let from = ''

  // Format the start date - only include year if it's in the past
  if (isPastYear) {
    from = start.toFormat("ccc MMM d, yyyy '•' t")
  } else {
    from = start.toFormat("ccc MMM d '•' t")
  }

  // Format the end date/time if provided
  if (endTime) {
    if (isSameDay) {
      // If same day, only show the end time
      to = end.toFormat('t')
    } else if (end.get('year') < now.get('year')) {
      // If end date is in a past year, include the year
      to = end.toFormat("MMM d, yyyy '•' t")
    } else {
      // Otherwise just month, day and time
      to = end.toFormat("MMM d '•' t")
    }

    to = returnAsObj ? to : ' - ' + to
  }

  return returnAsObj ? { from, to } : from + to
}

export function isDateInTheFuture (
  date : string | Date | DateTime | Object
) : boolean {
  return toDateTime(date) > dateTimeNow()
}

/**
 * Returns the month name from a date string or Date object
 * @param {string|Date} date - Date string or Date object
 * @param {boolean} short - Whether to return short month name (e.g. 'Jan' vs 'January')
 * @param {string} timezone - Optional timezone (defaults to local timezone)
 * @returns {string} Month name
 */
export function getMonthFromDate (
  date : string | Date | DateTime | Object,
  short? : boolean,
  timezone? : string
) : string {
  return toDateTime(date, timezone).toFormat(short ? 'MMM' : 'MMMM')
}

/**
 * Returns the day number from a date string or Date object
 * @param {string|Date} date - Date string or Date object
 * @param {string} timezone - Optional timezone (defaults to local timezone)
 * @returns {number} Day of month (1-31)
 */
export function getDayFromDate (
  date : string | Date | DateTime | Object,
  timezone? : string
) : number {
  return toDateTime(date, timezone).day
}

/**
 * Returns the hour from a date string or Date object
 * @param {string|Date} date - Date string or Date object
 * @param {boolean} use24Hour - Whether to use 24-hour format (default: false)
 * @param {string} timezone - Optional timezone (defaults to local timezone)
 * @returns {string} Formatted hour (with AM/PM if use24Hour is false)
 */
export function getHourFromDate (
  date : string | Date | DateTime | Object,
  use24Hour? : boolean,
  timezone? : string
) : string {
  return toDateTime(date, timezone).toFormat(use24Hour ? 'HH' : 'h a')
}
