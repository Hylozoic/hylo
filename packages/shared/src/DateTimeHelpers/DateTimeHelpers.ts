import { DateTime, DateTimeUnit } from 'luxon'
import prettyDate from 'pretty-date'

export const getLocaleAsString = (locale : string ) : string => {
  switch (locale) {
    case 'en':
      return 'en-US'
    case 'es':
      return 'es'
    default:
      return 'en-US'
  }
}

export const dateTimeNow = (locale?: string) : DateTime => {
  return DateTime.now().setLocale(getLocaleAsString(locale || ''))
}

export interface ToDateTimeOptions {
  timezone?: string;
  locale?: string;
}

export const toDateTime = (
  dt: string | Date | DateTime | Object,
  options?: ToDateTimeOptions
): DateTime => {
  const { timezone, locale } = options || {};
  const zoneOption = { zone: timezone || DateTime.now().zoneName || 'UTC' }
  const _dt = dt instanceof DateTime
    ? dt
    : dt instanceof Date
      ? DateTime.fromJSDate(dt, zoneOption).setLocale(getLocaleAsString(locale || ''))
      : typeof dt === 'string'
        ? DateTime.fromISO(dt, zoneOption).setLocale(getLocaleAsString(locale || ''))
        : DateTime.fromObject(dt, zoneOption).setLocale(getLocaleAsString(locale || ''))
  if (_dt.invalidReason) {
    throw new Error(`Invalid date: ${_dt.invalidReason}`)
  }
  return _dt
}

const isSame = (
  dt1: string | Date | DateTime | Object,
  dt2: string | Date | DateTime | Object,
  unit: DateTimeUnit,
  locale?: string
): boolean => {
  const _dt1 = toDateTime(dt1, { locale })
  const _dt2 = toDateTime(dt2, { locale })
  return _dt1.hasSame(_dt2, unit)
}

export const rangeIncludesDate = (
  start: string | Date | DateTime | Object,
  date: string | Date | DateTime | Object,
  end: string | Date | DateTime | Object,
): boolean => {
  const _start = toDateTime(start)
  const _date = toDateTime(date)
  const _end = toDateTime(end)
  return _date.hasSame(_start, 'day')
    || _date.hasSame(_end, 'day')
    || (_start < _date && _date < _end)
}

export const inWeek = (
  start: string | Date | DateTime | Object,
  date: string | Date | DateTime | Object,
  end: string | Date | DateTime | Object
): boolean => {
  const _start = toDateTime(start)
  const _date = toDateTime(date)
  const _end = toDateTime(end)
  const weekStart = _date.startOf('week', { useLocaleWeeks: true })
  const weekEnd = _date.endOf('week', { useLocaleWeeks: true }).plus({ days: 1 })
  return _start < weekEnd && weekStart <= _end
}

export const isSameDay = (
  date1 : string | Date | DateTime | Object,
  date2 : string | Date | DateTime | Object
) : boolean => {
  return isSame(date1, date2, 'day')
}

export const isSameWeek = (
  date1 : string | Date | DateTime | Object,
  date2 : string | Date | DateTime | Object
) : boolean => {
  return isSame(date1, date2, 'week')
}

export const isSameMonth = (
  date1 : string | Date | DateTime | Object,
  date2 : string | Date | DateTime | Object
) : boolean => {
  return isSame(date1, date2, 'month')
}

export function humanDate (
  date: string | Date | DateTime | Object,
  short?: boolean,
  locale?: string
): string {
  const _date = toDateTime(date, { locale })
  let ret = _date.invalidReason ? '' : prettyDate.format(_date.toJSDate())

  // Always return 'now' for very recent timestamps
  if (ret === 'just now') {
    return 'now'
  }

  if (short) {
    ret = ret.replace(' ago', '')
  } else {
    if (ret.match(/(\d+) seconds? ago/)) {
      return 'now'
    }
  }

  return ret.replace(/ seconds?/, 's')
    .replace(/ minutes?/, 'm')
    .replace(/ hours?/, 'h')
    .replace(/ days?/, 'd')
    .replace(/ weeks?/, 'w')
    .replace(/ years?/, 'y')
    .replace(/ month(s?)/, ' mo$1')
}

export interface FormatDatePairOptions {
  start: string | Date | DateTime | Object,
  end?: string | Date | DateTime | Object | boolean,
  returnAsObj?: boolean,
  timezone?: string,
  locale?: string,
  skipTime?: boolean
}

export const formatDatePair = ({
  start,
  end,
  returnAsObj = false,
  timezone,
  locale,
  skipTime = false
}: FormatDatePairOptions): string | { from: string, to: string } => {
  const _start = toDateTime(start, { timezone, locale })
  const _end = end ? toDateTime(end, { timezone, locale }) : null
  const now = dateTimeNow(locale)

  // Base formats without timezone
  const formatWithYear = skipTime ? 'ccc MMM d, yyyy' : "ccc MMM d, yyyy '•' t"
  const formatWithoutYear = skipTime ? 'ccc MMM d' : "ccc MMM d '•' t"

  // Formats with timezone
  const formatWithYearWithTz = skipTime ? 'ccc MMM d, yyyy' : "ccc MMM d, yyyy '•' t ZZZZ"
  const formatWithoutYearWithTz = skipTime ? 'ccc MMM d' : "ccc MMM d '•' t ZZZZ"

  const isSameDay = _end && _start.get('day') === _end.get('day') &&
                    _start.get('month') === _end.get('month') &&
                    _start.get('year') === _end.get('year')

  let to = ''
  let from = ''

  // Format the start date - only include year if it's not this year
  // Include the timezone if the end date is not provided
  if (_start.get('year') !== now.get('year')) {
    from = _start.toFormat(end ? formatWithYear : formatWithYearWithTz)
  } else {
    from = _start.toFormat(end ? formatWithoutYear : formatWithoutYearWithTz)
  }

  // Format the end date/time if provided
  if (_end) {
    if (isSameDay) {
      // If same day, only show the end time (with timezone if not skipping time)
      to = _end.toFormat(skipTime ? 't' : 't ZZZZ')
    } else if (_end.get('year') < now.get('year')) {
      // If end date is in a past year, include the year (with timezone)
      to = _end.toFormat(formatWithYearWithTz)
    } else {
      // Otherwise just month, day and time (with timezone)
      to = _end.toFormat(formatWithoutYearWithTz)
    }
    to = returnAsObj ? to : ' - ' + to
  }

  return returnAsObj ? { from, to } : from + to
}

export function isDateInTheFuture (
  date: string | Date | DateTime | Object,
  locale?: string
): boolean {
  return toDateTime(date, { locale }) > dateTimeNow(locale)
}

/**
 * Returns the month name from a date string or Date object
 * @param {string|Date} date - Date string or Date object
 * @param {boolean} short - Whether to return short month name (e.g. 'Jan' vs 'January')
 * @param {string} timezone - Optional timezone (defaults to local timezone)
 * @param {string} locale - Optional locale for formatting
 * @returns {string} Month name
 */
export function getMonthFromDate (
  date: string | Date | DateTime | Object,
  short?: boolean,
  timezone?: string,
  locale?: string
): string {
  return toDateTime(date, { timezone, locale }).toFormat(short ? 'MMM' : 'MMMM')
}

/**
 * Returns the day number from a date string or Date object
 * @param {string|Date} date - Date string or Date object
 * @param {string} timezone - Optional timezone (defaults to local timezone)
 * @param {string} locale - Optional locale for formatting
 * @returns {number} Day of month (1-31)
 */
export function getDayFromDate (
  date: string | Date | DateTime | Object,
  timezone?: string,
): number {
  return toDateTime(date, { timezone }).day
}

/**
 * Returns the hour from a date string or Date object
 * @param {string|Date} date - Date string or Date object
 * @param {boolean} use24Hour - Whether to use 24-hour format (default: false)
 * @param {string} timezone - Optional timezone (defaults to local timezone)
 * @param {string} locale - Optional locale for formatting
 * @returns {string} Formatted hour (with AM/PM if use24Hour is false)
 */
export function getHourFromDate (
  date: string | Date | DateTime | Object,
  use24Hour?: boolean,
  timezone?: string
): string {
  return toDateTime(date, { timezone }).toFormat(use24Hour ? 'HH' : 'h a')
}