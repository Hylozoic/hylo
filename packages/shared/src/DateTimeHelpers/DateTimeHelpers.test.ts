// @jest-environment node
// This test file is intended to be run with Jest, which provides the describe, it, and expect globals.
import * as DateTimeHelpers from './DateTimeHelpers'
import { DateTime } from 'luxon'

describe('formatDatePair', () => {
  it('displays differences of dates', () => {
    const d1 = DateTime.fromMillis(1551908483315, {zone: 'Etc/GMT'}).set({month: 1, day: 1, hour: 18})
    const d2 = d1.set({hour: 21})
    const d3 = d2.set({day: 2})
    const d4 = d3.set({month: 2})
    const d5 = d4.set({year: 2050})

    expect(DateTimeHelpers.formatDatePair({ start: d1, end: d2 })).toMatchSnapshot()
    expect(DateTimeHelpers.formatDatePair({ start: d1, end: d3 })).toMatchSnapshot()
    expect(DateTimeHelpers.formatDatePair({ start: d1, end: d4 })).toMatchSnapshot()
    expect(DateTimeHelpers.formatDatePair({ start: d1, end: d5 })).toMatchSnapshot()
    expect(DateTimeHelpers.formatDatePair({ start: d1, end: '' })).toMatchSnapshot()
  })

  it('can accept a custom timezone', () => {
    const d1 = DateTime.fromMillis(1551908483315, {zone: 'Etc/GMT'}).set({month: 1, day: 1, hour: 18})
    const d2 = d1.set({hour: 21})
    expect(DateTimeHelpers.formatDatePair({ start: d1, end: d2, timezone: 'America/New_York' })).toMatchSnapshot()
  })

  it('supports skipTime parameter', () => {
    const d1 = DateTime.fromMillis(1551908483315, {zone: 'Etc/GMT'}).set({month: 1, day: 1, hour: 18})
    const d2 = d1.set({hour: 21})
    const d3 = d2.set({day: 2})
    expect(DateTimeHelpers.formatDatePair({ start: d1, end: d2, skipTime: true })).toMatchSnapshot()
    expect(DateTimeHelpers.formatDatePair({ start: d1, end: d3, skipTime: true })).toMatchSnapshot()
  })

  it('returns object when returnAsObj is true', () => {
    const d1 = DateTime.fromMillis(1551908483315, {zone: 'Etc/GMT'}).set({month: 1, day: 1, hour: 18})
    const d2 = d1.set({hour: 21})
    expect(DateTimeHelpers.formatDatePair({ start: d1, end: d2, returnAsObj: true })).toMatchSnapshot()
  })
})
  