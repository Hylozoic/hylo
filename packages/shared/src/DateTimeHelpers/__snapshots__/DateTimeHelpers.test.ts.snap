// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatDatePair can accept a custom timezone 1`] = `"Tue Jan 1, 2019 • 6:41 PM - 9:41 PM UTC"`;

exports[`formatDatePair displays differences of dates 1`] = `"Tue Jan 1, 2019 • 6:41 PM - 9:41 PM UTC"`;

exports[`formatDatePair displays differences of dates 2`] = `"Tue Jan 1, 2019 • 6:41 PM - Wed Jan 2, 2019 • 9:41 PM UTC"`;

exports[`formatDatePair displays differences of dates 3`] = `"Tue Jan 1, 2019 • 6:41 PM - Sat Feb 2, 2019 • 9:41 PM UTC"`;

exports[`formatDatePair displays differences of dates 4`] = `"Tue Jan 1, 2019 • 6:41 PM - Wed Feb 2 • 9:41 PM UTC"`;

exports[`formatDatePair displays differences of dates 5`] = `"Tue Jan 1, 2019 • 6:41 PM UTC"`;

exports[`formatDatePair returns object when returnAsObj is true 1`] = `
Object {
  "from": "Tue Jan 1, 2019 • 6:41 PM",
  "to": "9:41 PM UTC",
}
`;

exports[`formatDatePair supports skipTime parameter 1`] = `"Tue Jan 1, 2019 - 9:41 PM"`;

exports[`formatDatePair supports skipTime parameter 2`] = `"Tue Jan 1, 2019 - Wed Jan 2, 2019"`;
